src/lxc/network.c

```diff
@@ -2679,6 +2717,151 @@ bool lxc_delete_network_unpriv(struct lxc_handler *handler)
 	return true;
 }
 
+struct ip_proxy_args {
+	const char *ip;
+	const char *dev;
+};
+
+static int lxc_add_ip_proxy_exec_wrapper(void *data)
+{
+	struct ip_proxy_args *args = data;
+
+	execlp("ip", "ip", "neigh", "add", "proxy", args->ip, "dev", args->dev,
+	       (char *)NULL);
+	return -1;
+}
+
+static int lxc_del_ip_proxy_exec_wrapper(void *data)
+{
+	struct ip_proxy_args *args = data;
+
+	execlp("ip", "ip", "neigh", "flush", "proxy", args->ip, "dev", args->dev,
+	       (char *)NULL);
+	return -1;
+}
+
+static int lxc_add_ip_proxy(const char *ip, const char *dev)
+{
+	int ret;
+	char cmd_output[PATH_MAX];
+	struct ip_proxy_args args;
+	args.ip = ip;
+	args.dev = dev;
+
+	ret = run_command(cmd_output, sizeof(cmd_output),
+			  lxc_add_ip_proxy_exec_wrapper, (void *)&args);
+	if (ret < 0) {
+		ERROR("Failed to add ip proxy \"%s\" to dev \"%s\": %s", ip, dev, cmd_output);
+		return -1;
+	}
+
+	return 0;
+}
+
+static int lxc_del_ip_proxy(const char *ip, const char *dev)
+{
+	int ret;
+	char cmd_output[PATH_MAX];
+	struct ip_proxy_args args;
+	args.ip = ip;
+	args.dev = dev;
+
+	ret = run_command(cmd_output, sizeof(cmd_output),
+			  lxc_del_ip_proxy_exec_wrapper, (void *)&args);
```

Comment:
The cast to void is not needed.

Reply:
@brauner will do,  see https://github.com/lxc/lxc/pull/2964/files#r280346212